cesium/internal/unary/db.go

```diff
@@ -83,13 +88,63 @@ func (db *DB) OpenIterator(cfg IteratorConfig) *Iterator {
 	cfg = DefaultIteratorConfig.Override(cfg)
 	iter := db.Domain.NewIterator(cfg.ranger())
 	i := &Iterator{
-		idx:            db.index(),
-		Channel:        db.Channel,
-		internal:       iter,
-		IteratorConfig: cfg,
+		idx:              db.index(),
+		Channel:          db.Channel,
+		internal:         iter,
+		IteratorConfig:   cfg,
+		decrementCounter: func() { db.openIteratorWriters.Add(-1) },
```

Comment:
This is my stylistic preference, but I would generally name this function `onClose` because that makes it clearer that it should only be called when the iterator is closed, and it could be expanded to do other things in the future (such as end a trace) instead of just decrementing a counter.